### PR TITLE
allow escaped forward-slashes

### PR DESCRIPTION
**Pull request by dpopp07**
Context:
I am using `js-yaml` to load Swagger files in my app. Today, my app crashed and I discovered it was because a string had an escaped forward-slash (`\/`). My stack trace:
```
{ while scanning a double-quoted scalar
  on line 487, column 1
found unknown escape character /
  on line 487, column 28
    at ScannerError.YAMLError [as constructor] (myProject/node_modules/yaml-js/lib/errors.js:70:46)
    at ScannerError.MarkedYAMLError [as constructor] (myProject/node_modules/yaml-js/lib/errors.js:90:45)
    at new ScannerError (myProject/node_modules/yaml-js/lib/scanner.js:23:49)
    at Loader.__dirname.Scanner.Scanner.scan_flow_scalar_non_spaces (myProject/node_modules/yaml-js/lib/scanner.js:1203:19)
    at Loader.__dirname.Scanner.Scanner.scan_flow_scalar (myProject/node_modules/yaml-js/lib/scanner.js:1152:37)
    at Loader.__dirname.Scanner.Scanner.fetch_flow_scalar (myProject/node_modules/yaml-js/lib/scanner.js:562:36)
    at Loader.__dirname.Scanner.Scanner.fetch_double (myProject/node_modules/yaml-js/lib/scanner.js:556:19)
    at Loader.__dirname.Scanner.Scanner.fetch_more_tokens (myProject/node_modules/yaml-js/lib/scanner.js:235:21)
    at Loader.__dirname.Scanner.Scanner.check_token (myProject/node_modules/yaml-js/lib/scanner.js:115:14)
    at Loader.__dirname.Parser.Parser.parse_flow_sequence_entry (myProject/node_modules/yaml-js/lib/parser.js:481:18)
  context: 'while scanning a double-quoted scalar',
```

While forward slashes don't really need to be escaped, it is [allowed by the JSON spec](https://tools.ietf.org/html/rfc7159#section-7), so I don't think that `yaml-js` should crash when this occurs.

This PR adds very simple support to resolve the issue and I verified that the changes work in my app.

### CODE REVIEW

File: lib/scanner.js [hunk 1]
```diff
@@ -1536,7 +1536,8 @@
       'N': '\x85',
       '_': '\xA0',
       'L': '\u2028',
-      'P': '\u2029'
+      'P': '\u2029',
```
Comment (kylef) [inline lib/scanner.js]: This file is compiled from https://github.com/connec/yaml-js/blob/579e6f6cc513ded1740959fc57aac3dc90c5f788/src/lib/scanner.coffee#L41. So the source file should be changed.

Comment (dpopp07) [reply lib/scanner.js]: Good call, thanks. The source file has been updated

**Comment by connec** [comment]
This looks like one of the few incompatibilities between YAML 1.1 and YAML 1.2. The spec for YAML 1.2 was adjusted so that JSON is a proper subset, including this point (treating `\/` as a valid escape sequence) as well as the [valid line break characters](http://yaml.org/spec/1.2/spec.html#id2774608) (interestingly the missing escape sequence is not mentioned in the incompatibilities in the 1.2 spec).

As such, I'd rather not merge this without figuring out if/how the library should work with YAML 1.1 vs. YAML 1.2, which I'll try to consider soon. The list of differences seems small enough that the work involved should be reasonably low.

For now the, rather unsatisfactory, fact is that `\/` is technically invalid YAML 1.1 and so the parser behaviour is correct, so your source file will need to be changed.

**Comment by connec** [comment]
I'm going to close this PR, copy my comment to the issue, and leave that open to track this. Thanks for the report and the proposed fix!

**Pull request closed without merging**